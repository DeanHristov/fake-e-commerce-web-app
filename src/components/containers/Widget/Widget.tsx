'use client';

import { FC, ReactNode } from 'react';
import { useIsClient } from '../../../hooks/useIsClient';

type TStyle = 'white' | 'indigo';

export interface IWidgetProps {
  children: ReactNode;
  title?: string;
  subTitle?: string;
  header?: ReactNode;
  bg?: TStyle;
}

const Widget: FC<IWidgetProps> = ({ children, title, subTitle, header }) => {
  const isClient: boolean = useIsClient();

  if (!isClient) return null;

  return (
    <section className="bg-white h-auto m-auto border border-gray-300 rounded-lg p-4 shadow-lg">
      {/** Widget Header */}
      {header && header}
      {!header && (
        <div className="py-0">
          {title && <h3 className="text-base text-black font-bold">{title}</h3>}
          {subTitle && <h4 className="text-sm text-gray-600">{subTitle}</h4>}
        </div>
      )}

      {/** Widget Body/footer */}
      <div className="pt-2">{children}</div>
    </section>
  );
};

export default Widget;
