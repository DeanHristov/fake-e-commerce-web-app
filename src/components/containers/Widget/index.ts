import Widget, { IWidgetProps } from './Widget';

export type { IWidgetProps };

export default Widget;
