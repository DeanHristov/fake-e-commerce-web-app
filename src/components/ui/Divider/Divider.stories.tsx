import type { Meta, StoryObj } from '@storybook/react';

import Divider, { IDividerProps } from './index';

const meta: Meta<IDividerProps> = {
  title: 'UI/Divider',
  component: Divider,
  argTypes: {},
};

export default meta;

type Story = StoryObj<IDividerProps>;

export const Default: Story = {
  args: {},
};

export const WithGradient: Story = {
  args: {
    gradient: true,
    color: 'via-blue-500',
  },
};
