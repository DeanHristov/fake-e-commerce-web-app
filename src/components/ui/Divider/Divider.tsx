'use client';

import { FC } from 'react';

export interface IDividerProps {
  color?: string;
  abv?: boolean;
  gradient?: boolean;
}

const Divider: FC<IDividerProps> = ({ color, gradient = false }) => {
  const gradientStyles = `w-full h-px m-auto bg-gradient-to-r from-transparent ${
    color ?? 'via-gray-400'
  } to-transparent`;
  const defaultStyles = `w-full m-auto border-b ${color ?? 'border-gray-400'}`;

  return <div className={gradient ? gradientStyles : defaultStyles} />;
};

export default Divider;
