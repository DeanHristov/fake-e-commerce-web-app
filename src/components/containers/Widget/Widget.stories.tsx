import type { Meta, StoryObj } from '@storybook/react';

import Widget, { IWidgetProps } from './index';
import Divider from '../../ui/Divider';

const meta: Meta<IWidgetProps> = {
  title: 'Containers/Widget',
  component: Widget,
  parameters: {
    layout: 'centered',
  },
  argTypes: {},
};

export default meta;

type Story = StoryObj<IWidgetProps>;

export const Default: Story = {
  render: () => (
    <Widget
      title="Widget title"
      subTitle="Lorem ipsum dolor sit amet, consectetur"
    >
      <Divider gradient />
      <h3 className="leading-loose text-center text-slate-900">
        &#169; 2023 | Fake App
      </h3>
    </Widget>
  ),
};

export const WithoutFooter: Story = {
  render: () => (
    <Widget
      title="Widget title"
      subTitle="Lorem ipsum dolor sit amet, consectetur"
    >
      <div className="w-96">Body content...</div>
    </Widget>
  ),
};
