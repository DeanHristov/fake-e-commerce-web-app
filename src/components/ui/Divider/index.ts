import Divider, { IDividerProps } from './Divider';

export type { IDividerProps };

export default Divider;
