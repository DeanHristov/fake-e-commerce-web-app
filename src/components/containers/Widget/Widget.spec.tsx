import { render } from '@testing-library/react';
import Widget from './index';
import Divider from '../../ui/Divider';

describe('Container/Widget <Widget {...} />', () => {
  it('Should render default state', () => {
    const { container } = render(
      <Widget title="Widget title">
        <div className="w-96 pb-2">Body content...</div>
        <Divider gradient />
        <h3 className="leading-loose text-center text-slate-900">
          &#169; 2023 | Fake App
        </h3>
      </Widget>,
    );

    expect(container).toMatchSnapshot();
  });
});
