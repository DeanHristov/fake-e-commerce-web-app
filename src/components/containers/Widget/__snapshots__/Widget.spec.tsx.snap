// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container/Widget <Widget {...} /> Should render default state 1`] = `
<div>
  <section
    class="bg-white h-auto m-auto rounded-lg p-4 shadow-lg"
  >
    <div
      class="py-0"
    >
      <h3
        class="text-base text-black font-bold"
      >
        Widget title
      </h3>
    </div>
    <div
      class="pt-2"
    >
      <div
        class="w-96 pb-2"
      >
        Body content...
      </div>
      <div
        class="w-full h-px m-auto bg-gradient-to-r from-transparent via-gray-400 to-transparent"
      />
      <h3
        class="leading-loose text-center text-slate-900"
      >
        Â© 2023 | Fake App
      </h3>
    </div>
  </section>
</div>
`;
